eth2/operation_pool/src/lib.rs

```diff
@@ -888,15 +909,114 @@ mod tests {
+                    let att = signed_attestation(
+                        &bc.committee,
+                        bc.index,
+                        keypairs,
+                        i..i + step_size,
+                        slot,
+                        state,
+                        spec,
+                        if i == 0 { None } else { Some(0) },
+                    );
+                    op_pool.insert_attestation(att, state, spec).unwrap();
+                }
+            };
+
+            for committee in &committees {
+                assert_eq!(committee.committee.len(), target_committee_size);
+                // Attestations signed by only 2-3 validators
+                insert_attestations(committee, small_step_size);
+                // Attestations signed by 4+ validators
+                insert_attestations(committee, big_step_size);
+            }
+
+            let num_small = target_committee_size / small_step_size;
+            let num_big = target_committee_size / big_step_size;
+
+            assert_eq!(op_pool.attestations.read().len(), committees.len());
+            assert_eq!(
+                op_pool.num_attestations(),
+                (num_small + num_big) * committees.len()
+            );
+            assert!(op_pool.num_attestations() > max_attestations);
+
+            state.slot += spec.min_attestation_inclusion_delay;
+            let best_attestations = op_pool
+                .get_attestations(state, spec)
+                .expect("should have valid best attestations");
+            assert_eq!(best_attestations.len(), max_attestations);
+
+            let active_indices = state
+                .get_cached_active_validator_indices(RelativeEpoch::Current)
+                .unwrap();
+            let total_active_balance = state.get_total_balance(&active_indices, spec).unwrap();
+
+            // Used for asserting that rewards are in decreasing order.
+            let mut prev_reward = u64::max_value();
+
+            for att in &best_attestations {
+                // Code taken from attestation.rs, modified to sum over the attestation indices.
+                // using aggregation_bits instead of earliest_attestation_validators() for indices.
+                let indices =
+                    get_attesting_indices(state, &att.data, &att.aggregation_bits).unwrap();
+                let rewards = indices.iter().fold(0u64, |sum, validator_index| {
+                    let reward = get_base_reward(
+                        state,
+                        *validator_index as usize,
+                        total_active_balance,
+                        spec,
+                    )
+                    .unwrap()
+                        / spec.proposer_reward_quotient;
+                    sum + reward
+                });
+
+                // Check that rewards are in descreasing order
+                assert!(prev_reward >= rewards);
+
+                prev_reward = rewards;
+            }
+        }
```

Comment:
Hey @michaelsproul ! I had tried toying around with the `get_attesting_indices` function, seing that it was used in the `attestation.rs`.
The test was still failing, so but I don't see any reason to why it would fail in `attestation.rs` so I think it's coming from my test function.
My intuition would be that tests are failing due to me doing a simple sum over attestation indices reward? But then what should I be doing?